### ISSUE TITLE
Improve rendering of JSDoc comment text with displayparts

### ISSUE THREAD

**Issue by sandersn** [microsoft/TypeScript#43085]
#41877 improves parsing of jsdoc comment text as well as giving the language service the option of providing display parts for comments instead of text. Currently everything is just `"text"` except for links. However, with small tweaks to code like `getCommentDisplayParts` in services/jsDoc.ts, it would be possible to format `@param` names as `"parameterName"`, `@template` names as `"typeParameterName"`, etc. 

This could make display a lot prettier.

Note: depends on #41877 being merged.

**Comment by hantatsang** [comment]
I'd like to work on this issue, I'll watch for when #41877 is merged

**Comment by hantatsang** [comment]
Now awaiting #43312

**Comment by hantatsang** [comment]
Hi, I'm not sure if I understand the requirement for this issue correctly. Currently, for example, `@param` name has already been `param`, `@template` name is `template`, etc. Not sure why you want them to be `parameterName`, `typeParameterName`, etc.?

Just wanna confirm to make sure I'm not doing something out of scope. If I'm wrong please give me some pointers. Thanks in advance.
